app/controllers/users/totp_setup_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Users
+  class TotpSetupController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def new
+      if session[:otp_secret_key].nil?
+        session[:otp_secret_key] = current_user.generate_totp_secret
+      end
```

Comment:
Hmm.. we need to figure out this session persistence thing.  I would have though that session variables are deleted when a user signs out.  I opened an issue to investigate:
https://github.com/18F/identity-private/issues/484